src/lib.rs

```diff
@@ -479,6 +507,110 @@ impl DoubleArrayAhoCorasick {
         }
     }
 
+    /// Serializes the automaton into the output stream.
+    ///
+    /// # Arguments
+    ///
+    /// * `writer` - Output stream.
+    ///
+    /// # Errors
+    ///
+    /// `std::io::Error` is returned if it fails to write the data.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use daachorse::DoubleArrayAhoCorasick;
+    ///
+    /// let patterns = vec!["bcd", "ab", "a"];
+    /// let pma = DoubleArrayAhoCorasick::new(patterns).unwrap();
+    ///
+    /// let mut buffer = vec![];
+    /// pma.serialize_into(&mut buffer).unwrap();
+    /// ```
+    pub fn serialize_into<W: io::Write>(&self, mut writer: W) -> io::Result<()> {
```

Comment:
In Rust, `_into` has a special meaning.
https://doc.rust-lang.org/std/convert/trait.Into.html

Please rename it to `serialize`.